src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewEngines/CompositeViewEngine.cs

```diff
@@ -77,6 +85,14 @@ public ViewEngineResult GetView(string executingFilePath, string viewPath, bool
                 throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(viewPath));
             }
 
+            if (ViewEngines.Count == 0)
+            {
+                throw new InvalidOperationException(Resources.FormatViewEnginesAreRequired(
+                    typeof(MvcViewOptions).FullName,
+                    nameof(MvcViewOptions.ViewEngines),
+                    typeof(IViewEngine).FullName));
+            }
```

Comment:
Disagree w/ the contrast you're making. Note the `IRazorViewEngineFileProviderAccessor` name and how it's used.